src/Microsoft.DocAsCode.Common/Git/GitUtility.cs

```diff
@@ -112,22 +124,35 @@ private static GitDetail GetFileDetailCore(string filePath)
         private static GitRepoInfo GetRepoInfoCore(string directory)
         {
             var repoRootPath = RunGitCommandAndGetFirstLine(directory, GetRepoRootCommand);
-            var localBranch = GetLocalBranchName(repoRootPath);
-
+          
+            // Many build systems use a "detach head", which means that the normal git commands
+            // to get branch names do not work.  Thankfully, they set an environment variable.
+            string localBranch = BuildSystemBranchName
```

Comment:
The environment variable names are overrides.  If any is specified, then they specify the branch name.